pkg/shared/contextutil/util.go

```diff
@@ -109,3 +110,37 @@ func GetRegistryForServiceConfig(currCtx *servicecontext.ServiceConfig, f *facto
 
 	return &registryInstance, err
 }
+
+// GetCurrentConnectorInstance returns the connector instance set in the currently selected context
+func GetCurrentConnectorInstance(f *factory.Factory) (*connectormgmtclient.Connector, error) {
+
+	svcContext, err := f.ServiceContext.Load()
+	if err != nil {
+		return nil, err
+	}
+
+	currCtx, err := GetCurrentContext(svcContext, f.Localizer)
+	if err != nil {
+		return nil, err
+	}
+
+	return GetConnectorForServiceConfig(currCtx, f)
+}
+
+func GetConnectorForServiceConfig(currCtx *servicecontext.ServiceConfig, f *factory.Factory) (*connectormgmtclient.Connector, error) {
+	conn, err := f.Connection(connection.DefaultConfigSkipMasAuth)
```

Comment:
That would create 2 connections all the time - despite someone not even using connectors. 
I think we might pass connection to this method (entire context object) instead.